bee-message/src/payload/transaction/mod.rs

```diff
@@ -53,26 +48,105 @@ impl Transaction {
         &self.unlock_blocks
     }
 
-    pub fn builder(seed: &Seed) -> TransactionBuilder {
-        TransactionBuilder::new(seed)
+    pub fn builder() -> TransactionBuilder {
+        TransactionBuilder::default()
     }
+}
+
+impl Packable for Transaction {
+    fn packed_len(&self) -> usize {
+        self.essence.packed_len()
+            + 0u16.packed_len()
+            + self.unlock_blocks.iter().map(|block| block.packed_len()).sum::<usize>()
+    }
+
+    fn pack<W: Write>(&self, buf: &mut W) -> Result<(), PackableError> {
+        self.essence.pack(buf)?;
+
+        (self.unlock_blocks.len() as u16).pack(buf)?;
+        for unlock_block in self.unlock_blocks.as_ref() {
+            unlock_block.pack(buf)?;
+        }
+
+        Ok(())
+    }
+
+    fn unpack<R: Read + ?Sized>(buf: &mut R) -> Result<Self, PackableError>
+    where
+        Self: Sized,
+    {
+        let essence = TransactionEssence::unpack(buf)?;
+
+        let unlock_blocks_len = u16::unpack(buf)? as usize;
+        let mut unlock_blocks = Vec::with_capacity(unlock_blocks_len);
+        for _ in 0..unlock_blocks_len {
+            unlock_blocks.push(UnlockBlock::unpack(buf)?);
+        }
+
+        Ok(Self {
+            essence,
+            unlock_blocks: unlock_blocks.into_boxed_slice(),
+        })
+    }
+}
 
-    pub fn validate(&self) -> Result<(), Error> {
+#[allow(dead_code)]
+fn is_sorted<T: Ord>(iterator: Iter<T>) -> bool {
+    let mut iterator = iterator;
+    let mut last = match iterator.next() {
+        Some(e) => e,
+        None => return true,
+    };
+
+    for curr in iterator {
+        if let Ordering::Greater = &last.cmp(&curr) {
+            return false;
+        }
+        last = curr;
+    }
+
+    true
+}
+
+#[derive(Debug, Default)]
+pub struct TransactionBuilder {
+    essence: TransactionEssenceBuilder,
+    unlock_blocks: Vec<UnlockBlock>,
+}
+
+impl TransactionBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn with_essence(mut self, essence: TransactionEssenceBuilder) -> Self {
```

Comment:
Could it pass built `TransactionEssence` here? I could use `Packable` trait to get serialized essence before passing in this way.